app/scripts/controllers/network/network-controller.js

```diff
@@ -179,62 +198,106 @@ export default class NetworkController extends EventEmitter {
     if (EIPS[1559] !== undefined) {
       return EIPS[1559];
     }
-    const latestBlock = await this._getLatestBlock();
-    const supportsEIP1559 =
-      latestBlock && latestBlock.baseFeePerGas !== undefined;
-    this._setNetworkEIPSupport(1559, supportsEIP1559);
+    const supportsEIP1559 = await this._determineEIP1559Compatibility();
+    this.networkDetails.putState({
+      EIPS: {
+        ...this.networkDetails.getState().EIPS,
+        1559: supportsEIP1559,
+      },
+    });
     return supportsEIP1559;
   }
 
+  /**
+   * Captures information about the currently selected network — namely,
+   * the network ID and whether the network supports EIP-1559 — and then uses
+   * the results of these requests to determine the status of the network.
+   */
   async lookupNetwork() {
-    // Prevent firing when provider is not defined.
+    const { chainId } = this.providerStore.getState();
+    let networkChanged = false;
+    let networkId;
+    let supportsEIP1559;
+    let networkStatus;
+
     if (!this._provider) {
       log.warn(
         'NetworkController - lookupNetwork aborted due to missing provider',
       );
       return;
     }
 
-    const { chainId } = this.providerStore.getState();
     if (!chainId) {
       log.warn(
         'NetworkController - lookupNetwork aborted due to missing chainId',
       );
-      this._setNetworkState('loading');
-      this._clearNetworkDetails();
+      this._resetNetworkId();
+      this._resetNetworkStatus();
+      this._resetNetworkDetails();
       return;
     }
 
-    // Ping the RPC endpoint so we can confirm that it works
-    const initialNetwork = this.networkStore.getState();
-    const { type } = this.providerStore.getState();
-    const isInfura = INFURA_PROVIDER_TYPES.includes(type);
-
-    if (isInfura) {
-      this._checkInfuraAvailability(type);
-    } else {
-      this.emit(NETWORK_EVENTS.INFURA_IS_UNBLOCKED);
-    }
+    this.once(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
+      networkChanged = true;
+    });
 
-    let networkVersion;
-    let networkVersionError;
     try {
-      networkVersion = await this._getNetworkId();
+      const results = await Promise.all([
```

Comment:
There are quite a few changes to this method:

* Two requests are made in parallel to fetch the network ID and the latest block.
* The latter request (for the latest block) will be used to determine whether the network supports EIP-1559.
* If both requests succeed, then we assume that the network is available, but if either request fails, then we cannot make that determination.
* If a request fails, Infura may be blocking requests in general. If this happens, then Infura will return `{ error: 'countryBlocked' }` as the response body (JSON-decoded of course). Prior to these changes, if we knew that we were switching to an Infura network, we would make a direct request for `eth_blockNumber` (using `fetch`) and just test the response directly. We could of course keep doing this, but considering that we already use eth-query to talk to the network, it's better to keep doing that (since it handles errors showing up different ways). The downside to using eth-query, though, is that we now have to catch that error; and that means we have to figure out whether the error we've caught is a JSON-RPC error or something else.
* Previously, the availability check for an Infura network was offloaded to `_checkInfuraAvailability`, but this method was not awaited. That made things painful because it meant that any time we wanted to use `lookupNetwork` — or a method that called `lookupNetwork` — we had to wait for state to change before continuing. Now any subrequest we are making is awaited, i.e., there shouldn't be any async operations in `lookupNetwork` that aren't being awaited. This should clean up a lot of code down the line.